src/chocolatey.resources/helpers/functions/Get-UninstallRegistryKey.ps1

```diff
@@ -120,15 +119,20 @@ param(
     } catch {
       Write-Debug "Found bad key."
       foreach ($key in $keys){ try { Get-ItemProperty $key.PsPath > $null } catch { $badKey = $key.PsPath }}
-      Write-Verbose "Skipping bad key: $($key.PsPath)"
-      [array]$keys = Get-ChildItem -Path @($machine_key6432, $machine_key, $local_key) `
-                                   -ErrorAction SilentlyContinue `
-                     | Where-Object { $badKey -NotContains $_.PsPath }
+      Write-Verbose "Skipping bad key: $badKey"
+      [array]$keys = $keys | Where-Object { $badKey -NotContains $_.PsPath }
     }
     
     if ($success) { break; }
+    if ($attempt -eq 10) {
+      Write-Warning "Found more than 10 bad registry keys. Run command again with `'--verbose --debug`' for details."
+      Write-Debug "Each key searched should correspond to an installed program. It is very unlikely to have more than a few programs with incorrectly encoded keys, if any at all. This may be indicative of one or more corrupted registry branches."
+    }
   }
   
+  if ($($foundkey.Count) -eq $null) {
+    Write-Verbose "No registry key found with SoftwareName:`'$softwareName`'";
```

Comment:
This one almost feels like Write-Host or Write-Warning. In this case the latter.